pytest failing on home machine, cannot run make check

After a recent update with the master, and running make check, I got the message:

     py.test is required to run the integration tests, please install using 'pip3 install -r tests/requirements.txt

Fair enough, I ran the above command and tried again.  I got the above message again.  

There was no pytest in my path, so this Makefile line wasn't finding anything:

    PYTEST := $(shell command -v pytest 2> /dev/null)

I added `~/.local/bin` to my path. Then running `make check` gave me:

<pre>
PYTHONPATH=contrib/pylightning:$PYTHONPATH TEST_DEBUG=1 DEVELOPER=0 /home/wythe/.local/bin/pytest -vx tests/ --test-group=1 --test-group-count=1 -v -x
usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --test-group=1 --test-group-count=1
  inifile: None
  rootdir: /home/wythe/work/temp/lightning
Makefile:213: recipe for target 'pytest' failed

</pre>

Obviously there is something dreadfully wrong with my environment.  And I'm not much of a python guy.  Any help getting me up and running would be appreciated.

I did all this from a fresh clone of master.

Thanks. I can run individual tests now, so I am back up and running. Things like this:

<pre>
PYTHONPATH=contrib/pylightning DEVELOPER=1 NO_VALGRIND=1 python3 tests/test_lightningd.py -f LightningDTests.test_funding_all
</pre>

But `make check` results in:

<pre>
=========================================================== test session starts ============================================================
platform linux2 -- Python 2.7.14, pytest-3.5.1, py-1.5.3, pluggy-0.6.0 -- /usr/bin/python
cachedir: .pytest_cache
rootdir: /home/wythe/work/lightning, inifile:
plugins: test-groups-1.0.3
collecting 0 items / 1 errors                                                                                                              
================================================================== ERRORS ==================================================================
__________________________________________________ ERROR collecting tests/test_closing.py __________________________________________________
../../.local/lib/python2.7/site-packages/_pytest/python.py:412: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
../../.local/lib/python2.7/site-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
../../.local/lib/python2.7/site-packages/_pytest/assertion/rewrite.py:213: in load_module
    py.builtin.exec_(co, mod.__dict__)
tests/test_closing.py:1: in <module>
    from fixtures import *  # noqa: F401,F403
E     File "/home/wythe/work/lightning/tests/fixtures.py", line 136
E       print(l, end='')
E                   ^
E   SyntaxError: invalid syntax
========================================================= 1 error in 0.04 seconds ==========================================================
Makefile:213: recipe for target 'pytest' failed
make[1]: *** [pytest] Error 1
make[1]: Leaving directory '/home/wythe/work/lightning'
Makefile:205: recipe for target 'check' failed
make: *** [check] Error 2
</pre>

Maybe a python version issues?  But I'm back and running so no rush.

--- a/Makefile
+++ b/Makefile
@@ -196,9 +196,8 @@ include devtools/Makefile
 # Git doesn't maintain timestamps, so we only regen if git says we should.
 CHANGED_FROM_GIT = [ x"`git log $@ | head -n1`" != x"`git log $< | head -n1`" -o x"`git diff $<`" != x"" ]
 
-ifeq ($(TEST_GROUP_COUNT),)
-TEST_GROUP=1
-TEST_GROUP_COUNT=1
+ifneq ($(TEST_GROUP_COUNT),)
+PYTEST_OPTS += --test-group=$(TEST_GROUP) --test-group-count=$(TEST_GROUP_COUNT)
 endif
 
 check:
@@ -210,7 +209,7 @@ ifndef PYTEST
 	@echo "py.test is required to run the integration tests, please install using 'pip3 install -r tests/requirements.txt'"
 	exit 1
 else
-	PYTHONPATH=contrib/pylightning:$$PYTHONPATH TEST_DEBUG=1 DEVELOPER=$(DEVELOPER) $(PYTEST) -vx tests/ --test-group=$(TEST_GROUP) --test-group-count=$(TEST_GROUP_COUNT) $(PYTEST_OPTS)
+	PYTHONPATH=contrib/pylightning:$$PYTHONPATH TEST_DEBUG=1 DEVELOPER=$(DEVELOPER) $(PYTEST) tests/ $(PYTEST_OPTS)
 endif
 
 # Keep includes in alpha order.

